usaspending_api/etl/management/commands/reveal_last_period_award_updates.py

```diff
@@ -0,0 +1,85 @@
+    INNER JOIN disaster_emergency_fund_code defc ON
+        defc.group_name = 'covid_19'
+        AND faba.disaster_emergency_fund_code = defc.code
+    INNER JOIN submission_attributes sa ON
+        faba.submission_id = sa.submission_id
+    INNER JOIN dabs_submission_window_schedule dabs ON
+        dabs.id = sa.submission_window_id
+        AND dabs.submission_fiscal_year = {year}
+        AND dabs.submission_fiscal_month = {month}
+        AND dabs.is_quarter = {is_quarter}
+    INNER JOIN awards a on faba.award_id = a.id
+);
+"""
+
+
+class Command(BaseCommand):
+
+    """
+    NOTE: This command is necessary to update award values in Elasticsearch that were modified in the
+    latest submission window when that window's submissions are revealed. Elasticsearch calculations for
+    an award's Covid values do not include data from a submission until its submission window's reveal
+    date has been passed. Because incremental loads in Elasticsearch compare the award's update_date field
+    to a rolling date when determining whether it should be recalculated, the data will not automatically
+    be refreshed after a submission window's reveal date. This command updates the update_date field on
+    awards modified by the latest revealed submission so the Elasticsearch incremental load will include
+    those awards. The command uses the external_data_load_date table to make sure each submission window
+    is only 'revealed' once.
+    """
+
+    def handle(self, *args, **options):
+        script_start_time = timezone.now()
+        periods = retrieve_recent_periods()
+
+        self.last_load_date = get_last_load_date("reveal_last_period_award_updates", default=script_start_time)
+
+        total_records_updated = 0
+
+        total_records_updated += self.reveal_period_updates_if_behind(periods["this_month"])
+        total_records_updated += self.reveal_period_updates_if_behind(periods["this_quarter"])
+
+        update_last_load_date("reveal_last_period_award_updates", script_start_time)
+
+        logger.info(f"Found {total_records_updated} award records to update in Elasticsearch")
+
+        # Return will be captured as stdout in Jenkins job
+        return str(total_records_updated)
+
+    def reveal_period_updates_if_behind(self, period):
+
+        records_updated = 0
+
+        year = period["year"]
+        month = period["month"]
+        is_quarter = period["is_quarter"]
+
+        is_quarter_str = "quarter" if is_quarter else "month"
+
+        if self.last_load_date is None or self.last_load_date < period["submission_reveal_date"]:
+            formatted_sql = REVEAL_AWARD_SQL.format(year=year, month=month, is_quarter=is_quarter)
+
+            with connection.cursor() as cursor:
+                cursor.execute(formatted_sql)
+                records_updated = cursor.rowcount
+
+        logger.info(f"Revealing {records_updated} awards from {is_quarter_str} - year: {year}, period: {month}")
```

Comment:
My thinking was that the verb `reveal` is being done to `award updates`, so the updates are being revealed. I thought that provided more information than `touch_last_period_awards`. 

I do see that I'm not completely consistent, such as the logged message above which could read as `Revealing {records_updated:,} award updates from `...

Does this make more sense? If you still think that `touch` should be used, I can refactor.

Reply:
The term "reveal" was introduced to the USAspending repo to provide a straightforward mechanism for hiding DABS data until the submission period due date passed. At that time, all of the submission data should be revealed across the API and website. I think it is confusing to use the term for File D awards since holding back File D awards is not desirable. While the associated File C data from a open submission period isn't included the award already exists in Elasticsearch.